package/gluon-mesh-vpn-wireguard/files/lib/gluon/mesh-vpn/wireguard_pubkey.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+INCLUDE_ONLY=1
+. /lib/netifd/proto/wireguard.sh
+
+ensure_key_is_generated vpn
+iface="$(lua -e "vpn = require 'gluon.mesh-vpn'; print(vpn.get_interface())")"
+uci get "network.$iface.private_key" | /usr/bin/wg pubkey
```

Comment:
You are using a Linux ifname as the UCI section name here - this is just wrong. The UCI section is always called `mesh_vpn`.

Reply:
Oops, I see you are also introducing a new section based on `vpn_core.get_interface()` in a new upgrade script. Please don't do this, use static section names instead.

Reply:
The wireguard.sh proto determines it's interface name based on the uci section name:

See here: https://github.com/openwrt/openwrt/blob/master/package/network/utils/wireguard-tools/files/wireguard.sh#L117

Reply:
Let's call that interface something like `wg_mesh` then.

Reply:
This would imply interface names that are mesh-vpn provider dependent. While this would not cause much problems for most of the code, I have no idea yet how to solve this for the script [/lib/gluon/mesh-vpn/iptables-mss.rules](https://github.com/freifunk-gluon/gluon/blob/master/package/gluon-iptables-clamp-mss-to-pmtu/files/lib/gluon/mesh-vpn/iptables-mss.rules#L2).

However, I would rather suggest to use `mesh_vpn` as interface name. This could be used for all mesh-vpn providers and would be descriptive as well.

Reply:
Done.